packages/firebase_storage/ios/Classes/FirebaseStoragePlugin.m

```diff
@@ -80,6 +84,68 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
   }
 }
 
+- (FIRStorage *)getStorage:(FlutterMethodCall *)call {
+  NSString *appName = call.arguments[@"app"];
+  NSString *bucketUrl = call.arguments[@"bucket"];
+  FIRApp *app;
+
+  if ([appName isEqual:[NSNull null]]) {
+    app = [FIRApp defaultApp];
+  } else {
+    app = [FIRApp appNamed:appName];
+  }
+
+  if ([bucketUrl isEqual:[NSNull null]]) {
+    if (app.options.storageBucket) {
+      bucketUrl = [app.options.storageBucket isEqualToString:@""]
+                      ? @""
+                      : [@"gs://" stringByAppendingString:app.options.storageBucket];
+    } else {
+      bucketUrl = nil;
+    }
+  }
+
+  NSURL *url = [NSURL URLWithString:bucketUrl];
+  if (!url) {
+    // Call storage constructor to raise proper exception.
+    [FIRStorage storageForApp:app URL:bucketUrl];
+  }
+
+  NSMutableDictionary *bucketMap = _storageMap[app.name];
+  if (!bucketMap) {
+    bucketMap = [[NSMutableDictionary alloc] init];
+    _storageMap[app.name] = bucketMap;
+  }
+
+  NSString *bucketName = [url host];
+  FIRStorage *storage = bucketMap[bucketName];
+  if (!storage) {
+    // Raises an exception if bucketUrl is invalid.
+    storage = [FIRStorage storageForApp:app URL:bucketUrl];
+    bucketMap[bucketName] = storage;
+  }
+
+  return storage;
+}
+
+- (void)setMaxDownloadRetryTime:(FlutterMethodCall *)call result:(FlutterResult)result {
+  NSNumber *time = call.arguments[@"time"];
+  storage.maxDownloadRetryTime = [time longLongValue] / 1000;
```

Comment:
If time is 500ms, you'd end up with 0 because of integer division. You should divide by 1000.0.